sw/device/silicon_creator/rom/doc/sigverify.md

```diff
@@ -0,0 +1,183 @@
+*   Convert back to get the encoded message, i.e. _sig<sup>e</sup> mod n_ (1 Montgomery multiplication with signature).
+
+### Encoded Message Check
+
+OpenTitan uses the RSASSA-PKCS1-v1\_5[^1] signature scheme with SHA2-256, 3072-bit keys, e=65537 (F4) and does not support any other hash algorithms, key lengths, or exponents.
+Thus, sigverify uses a hard-coded padding for checking the padding of the encoded message.
+The rest of the encoded message, i.e.
+the expected digest, is compared with the digest of the image being verified.
+Sigverify uses a hardened implementation[^3] to protect against fault injection attacks.
+Please see this document for more information.
+
+## Public APIs
+
+```
+/**
+ * Returns the key with the given ID.
+ *
+ * This function returns the key only if it can be used in the given life cycle
+ * state and is valid in OTP. OTP check is performed only if the device is in a
+ * non-test operational state (PROD, PROD_END, DEV, RMA).
+ */
+rom_error_t sigverify_rsa_key_get(uint32_t key_id, lifecycle_state_t lc_state,
+                                  const sigverify_rsa_key_t **key);
+
+
+/**
+ * Verifies an RSASSA-PKCS1-v1_5 signature.
+ *
+ * The actual implementation that is used (software or OTBN) is determined by
+ * the life cycle state of the device and the OTP value.
+ */
+rom_error_t sigverify_rsa_verify(const sigverify_rsa_buffer_t *signature,
+                                 const sigverify_rsa_key_t *key,
+                                 const hmac_digest_t *act_digest,
+                                 lifecycle_state_t lc_state,
+                                 uint32_t *flash_exec);
+
+/**
+ * Gets the usage constraints struct that is used for verifying a ROM_EXT.
+ *
+ * This function reads
+ * - The device identifier from the life cycle controller,
+ * - Creator and owner manufacturing states from the OTP,
+ * - The life cycle state from life cycle controller, and
+ * masks the fields of `usage_constraints` according to the given
+ * `selector_bits`.
+ */
+void sigverify_usage_constraints_get(
+    uint32_t selector_bits, manifest_usage_constraints_t *usage_constraints);
+
+```
+
+## Testing
+
+Unit tests will be used to verify the algorithmic components of the software implementation, i.e. modular exponentiation and encoded message check, key retrieval, and behavior under different OTP values and life cycle states.
+Functional tests will be used to verify both the software and the OTBN implementations on target hardware.
+Sigverify will also have adversarial and random tests.
+
+## References
+
+[^1]: [PKCS #1: RSA Cryptography Specifications Version 2.2](https://tools.ietf.org/html/rfc8017#section-8.2)
+[^2]: [Manifest Format](https://docs.opentitan.org/sw/device/silicon_creator/rom_ext/docs/manifest/)
+[^3]: [Hardened Encoded Message Check for RSA Signature Verification](https://docs.google.com/document/u/0/d/1kBvQV9B3hKIflwhVQjGAntPrRnYvnLeo3WEyfxcO1R4/edit)
```

Comment:
I changed resolving this link to a TODO.  That document isn't trivially convertible to markdown because it's littered with equations that the markdown converter can't handle.